include/envoy/http/header_map.h

```diff
@@ -111,6 +111,12 @@ class HeaderString {
   void setInteger(uint64_t value);
 
   /**
+   * Set the value of the string to a static string reference.
+   * @param static_value MUST point to static data.
```

Comment:
Personally I kind of like the "ByReference" suggestion. I will do a bunch of find/replace and switch "static" with "reference" when it comes to headers. Will update PR.